src/background/services/wallet/index.ts

```diff
@@ -1211,17 +1213,10 @@ export default class WalletService extends GenericService {
       await b.write();
     }
 
-    const newTXs = await this.exec('node', 'getTXByAddresses', addresses);
-
-    if (!newTXs.length) {
-      return transactions;
-    }
-
-    transactions = transactions.concat(newTXs);
-    return await this.getAllReceiveTXs(startDepth + ACCOUNT_DEPTH, endDepth + ACCOUNT_DEPTH, transactions);
+    return this.exec('node', 'getTXByAddresses', addresses, startDepth, endDepth);
```

Comment:
i think the use of `startDepth` and `endDepth` here might be incorrect. The `startDepth` and `endDepth` variables indicates depth of the account, while the params passed to `getTXByAddresses` is intended for starting and ending block height to request tx from. Currently, this is requesting txs for addresses `[0 - 24]` for blocks `[0 - 24]`, and then addresses `[25 - 49]` for blocks `[25 - 49]`. 

I think we probably want to request blocks like this:
1. start with `addresses[0:99]` (to match bob desktop lookahead)
2. request all txs from `addresses[0:99]` for `blocks[0:n]` (`n = latest`)
  2a. if there are >100 txs (just my best guess, could be more), hapi returns 413
    - make `n = latest / 2`; repeat step 2;
  - if there are <= 100 txs, hapi returns 200
    - if `n < latest`; repeat step 2 with `blocks[latest-n:latest]`
3. repeat step 1 until all response from step 2 is empty